File: frontend/src/objects/admin/afterSales/AdminTicketSubmission.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  AfterSalesRequestType,
  CurrencyCents,
  IsoDateTime,
  PersonName,
  UserRole,
} from '@/objects/core/SharedObjects'

export type AdminTicketSubmission = {
  requestType?: AfterSalesRequestType
  submittedByRole?: UserRole
  submittedByName?: PersonName
  expectedCompensationCents?: CurrencyCents
  submittedAt: IsoDateTime
}

File: backend/src/main/scala/services/admin/objects/AdminTicketSubmission.scala
package services.admin.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given
import services.order.objects.*
import services.auth.objects.*

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class AdminTicketSubmission(
    requestType: Option[AfterSalesRequestType],
    submittedByRole: Option[UserRole],
    submittedByName: Option[PersonName],
    expectedCompensationCents: Option[CurrencyCents],
    submittedAt: IsoDateTime,
)
object AdminTicketSubmission:
  given Encoder[AdminTicketSubmission] = deriveEncoder
  given Decoder[AdminTicketSubmission] = deriveDecoder
